In [104]:
import requests
from lxml import html
import lxml

In [120]:
#r = requests.get('http://lachouquette.ch/category/style/mode/')

In [121]:
#tree = html.fromstring(r.content)

In [122]:
#truc = tree.xpath('//a[@class="next page-numbers"]')

In [123]:
#liens = tree.xpath('//section[@id="main"]//div[@class="row"]')[0].xpath('.//h3//a')

In [124]:
#href = [item.xpath('.//@href') for item in liens]
#titre = [item.xpath('.//text()') for item in liens]

In [125]:
#auteurs = tree.xpath('//section[@id="main"]//span[@class="entry-meta"]//a//text()')

In [126]:
#auteurs

In [202]:
categories = ['life','style','interviews','travel']
#categories = ['travel']

In [203]:
liens = list()
href = list()
titre = list()
auteurs = list()

for category in categories:
    r = requests.get('http://lachouquette.ch/category/'+category)
    tree = html.fromstring(r.content)
    liens = liens + tree.xpath('//section[@id="main"]//div[@class="row"]')[0].xpath('.//h3//a')
    href = href + [item.xpath('.//@href') for item in liens]
    titre = titre + [item.xpath('.//text()') for item in liens]
    auteurs = auteurs + tree.xpath('//section[@id="main"]//span[@class="entry-meta"]//a//text()')
    nextpage = tree.xpath('//section[@id="main"]//a[@class="next page-numbers"]')
    while len(nextpage):
        print(category)
        r = requests.get(nextpage[0].xpath('.//@href')[0])
        tree = html.fromstring(r.content)
        liens_while = tree.xpath('//section[@id="main"]//div[@class="row"]')[0].xpath('.//h3//a')
        liens = liens + liens_while
        href = href + [item.xpath('.//@href') for item in liens_while]
        titre = titre + [item.xpath('.//text()') for item in liens_while]
        auteurs = auteurs + tree.xpath('//section[@id="main"]//span[@class="entry-meta"]//a//text()')
        nextpage = tree.xpath('//section[@id="main"]//a[@class="next page-numbers"]')

    
    
    
    

life
life
life
life
life
life
life
life
life
life
life
style
style
style
style
interviews
interviews
travel


In [204]:
import pandas as pd

In [205]:
tab = pd.DataFrame([[item[0] for item in href],[item[0] for item in titre],auteurs])

In [206]:
tab = tab.transpose()

In [207]:
tab.columns = ['link','title','author']

In [213]:
print(len(tab['title'].unique()))
print(len(tab['author'].unique()))

320
11


# Résultat intermédiaire
Le nombre d'articles publiés est de 320 et le nombre d'auteurs est de 11 dans le blog de la Chouquette.